helm-charts/splunk-otel-collector/templates/daemonset.yaml

```diff
@@ -251,11 +251,7 @@ spec:
         imagePullPolicy: {{ .Values.image.otelcol.pullPolicy }}
         {{- if or $agent.securityContext (and (eq (include "splunk-otel-collector.logsEnabled" $) "true") (eq .Values.logsEngine "otel")) }}
         securityContext:
-          {{- if $agent.securityContext }}
-          {{- toYaml $agent.securityContext | nindent 10 }}
-          {{- else }}
-          runAsUser: 0
-          {{- end }}
+          {{- include "splunk-otel-collector.securityContext" (dict "isWindows" .Values.isWindows "securityContext" $agent.securityContext) | nindent 10 }}
```

Comment:
I don't think you have to pass `.Values.isWindows`. `$.Values.isWindows` should be available there

Reply:
@dmitryax 
If the caller is just passing the variable `.` then this would work, but I need to explicitly pass the securityContext of that particular block (ex: `$agent.securityContext`), so in helper func `.` will not have `.Values.isWindows` 

Is there a trick I am not aware of?

Reply:
I mean `$.Values.isWindows` not `.Values.isWindows`

Reply:
I can't access the `$.Values.isWindows` from the helper function, unless I pass `.` when I call the func
```<$.Values.isWindows>: nil pointer evaluating interface```